From dc2d4dae1df45be1d0a6f27b1bd3edf72ae355e2 Mon Sep 17 00:00:00 2001
From: Thiago Miranda <tmarques@windriver.com>
Date: Mon, 22 Dec 2025 18:38:00 -0300
Subject: [PATCH] Add backend checks to skip Ceph init for NetApp storage

Introduce conditional logic to skip Ceph initialization when using
NetApp backends (file, cinder, or pvc).
The script now prints an informational message and only runs the Ceph
bootstrap when the backend is rbd.

Signed-off-by: Thiago Miranda <tmarques@windriver.com>
---
 glance/templates/bin/_storage-init.sh.tpl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glance/templates/bin/_storage-init.sh.tpl b/glance/templates/bin/_storage-init.sh.tpl
index f780ff9e8..53ba9f155 100644
--- a/glance/templates/bin/_storage-init.sh.tpl
+++ b/glance/templates/bin/_storage-init.sh.tpl
@@ -30,8 +30,8 @@ if [[ "$SCHEME" == "https" && -f /etc/ssl/certs/openstack-helm.crt ]]; then
 fi
 
 set -ex
-if [ "x$STORAGE_BACKEND" == "xpvc" ]; then
-  echo "No action required."
+if [ "x$STORAGE_BACKEND" == "xfile" ] || [ "x$STORAGE_BACKEND" == "xpvc" ]; then
+  echo "INFO: No action required to use $STORAGE_BACKEND backend."
 elif [ "x$STORAGE_BACKEND" == "xswift" ]; then
   : ${OS_INTERFACE:="internal"}
   OS_TOKEN="$(openstack token issue -f value -c id)"
-- 
2.34.1

